tools/base/fpgainfo/board.c

```diff
@@ -0,0 +1,413 @@
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of  source code  must retain the  above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+
+#include <getopt.h>
+#include "fpgainfo.h"
+#include <opae/fpga.h>
+#include <wchar.h>
+#include <dirent.h>
+#include <string.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <dlfcn.h>
+#include "safe_string/safe_string.h"
+
+#include "board.h"
+
+
+// Board plug-in table
+static platform_data platform_data_table[] = {
+	{ 0x8086, 0xbcbd, "", NULL },
+	{ 0x8086, 0xbcc0, "", NULL },
+	{ 0x8086, 0xbcc1, "", NULL },
+	{ 0x8086, 0x09c4, "libboard_rc.so", NULL },
+	{ 0x8086, 0x09c5, "libboard_rc.so", NULL },
+	{ 0,      0,          NULL, NULL },
+};
+
+
+
+
+fpga_result load_board_plugin(fpga_token token, void** dl_hanlde)
+{
+	fpga_result res                = FPGA_OK;
+	fpga_properties props          = NULL;
+	uint16_t vendor_id             = 0;
+	uint16_t device_id             = 0;
+	int i                          = 0;
+
+	if (token == NULL || dl_hanlde == NULL) {
+		OPAE_ERR("Invalid input parameter");
+	}
+
+	res = fpgaGetProperties(token, &props);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get properties\n");
+		return false;
+	}
+
+	res = fpgaPropertiesGetDeviceID(props, &device_id);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get device ID\n");
+		goto err_out_destroy;
+	}
+
+	res = fpgaPropertiesGetVendorID(props, &vendor_id);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get vendor ID\n");
+		goto err_out_destroy;
+	}
+

```

Comment:
Need to lock a mutex around these accesses to platform_data_table to make them thread safe.

Reply:
fixed